File: Complementar - Estruturas de dados (Didático)/Algoritmos09/Alg09.py
''' 
♦ Algoritmo 09: Algoritmo de Pesquisa/Busca para Estruturas de Dados Lineares Estáticas (não Dinâmicas): Busca Sequencial (sem a utilização de variável/sentinela).
'''

def busca_sequencial(array, valor_procurado):
    for i in range(len(array)):
        if array[i] == valor_procurado:
            return i  # Retorna o índice do valor procurado
    return -1  # Retorna -1 se o valor não foi encontrado

array = [10, 5, 2, 8, 3, 7, 1, 6, 9, 4]
valor_procurado = 7

indice_encontrado = busca_sequencial(array, valor_procurado)

if indice_encontrado != -1:
    print(f"Valor {valor_procurado} encontrado no índice: {indice_encontrado}")
else:
    print(f"Valor {valor_procurado} não encontrado no array.")

File: Complementar - Estruturas de dados (Didático)/Algoritmos09/Alg09.js
/*
♦ Algoritmo 09: Algoritmo de Pesquisa/Busca para Estruturas de Dados Lineares Estáticas (não Dinâmicas): Busca Sequencial (sem a utilização de variável/sentinela).
*/

function buscaSequencial(array, valorProcurado) {
    for (let i = 0; i < array.length; i++) {
        if (array[i] === valorProcurado) {
            return i; // Retorna o índice do valor procurado
        }
    }
    return -1; // Retorna -1 se o valor não foi encontrado
}

const array = [10, 5, 2, 8, 3, 7, 1, 6, 9, 4];
const valorProcurado = 7;

const indiceEncontrado = buscaSequencial(array, valorProcurado);

if (indiceEncontrado !== -1) {
    console.log(`Valor ${valorProcurado} encontrado no índice: ${indiceEncontrado}`);
} else {
    console.log(`Valor ${valorProcurado} não encontrado no array.`);
}
